### ISSUE TITLE
Include fetch instructions in the "Submitted as pull.<num>.git.*" message

### ISSUE THREAD

**Issue by phil-blain** [gitgitgadget/gitgitgadget#274]
It would be nice if when the bot answers to a `/submit` command with the link to the submitted version on the list,
instructions for how to fetch that version locally would also be included (similar to the `Fetch-It-Via` footer of the cover letter in the mailing list submission).

I was thinking something along the lines of 
> To fetch this version into `FETCH_HEAD`:
>
>     git fetch https://github.com/gitgitgadget/git pr-<num>/<github_username>/<branch-name>-v<x>
>
> To fetch this version to a local tag:
>
>     git fetch https://github.com/gitgitgadget/git tag pr-<num>/<github_username>/<branch-name>-v<x> --no-tags
>

---

I wish the second command could be simply `git fetch <ggg url> tag <tag>`, but either the [`git fetch tag <tag>`](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt-ltrefspecgt) or the [`-no-tags`](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---no-tags) documentation is wrong, [or the doc is correct but the code is buggy...](https://lore.kernel.org/git/12F0A2A4-CC71-4452-B8B8-F540339F8519@gmail.com/T/#u)

**Comment by dscho** [comment]
> I wish the second command could be simply `git fetch <ggg url> tag <tag>`

I figure you meant only `<tag>` not `tag <tag>`. And yes, according to @peff's feedback, that works, right?

@phil-blain If you want to give it a try, you will probably want to touch [these lines](https://github.com/gitgitgadget/gitgitgadget/blob/e8ec25f755cc0b770ba956e2932f45b4d0fe0e03/lib/ci-helper.ts#L605-L607).

**Comment by phil-blain** [comment]
Yes it works correctly, the documentation for `--no-tags` could need a little clarification. I really meant `tag <tag>`, i.e. not have to also add `--no-tags`. (I guess `git fetch <tag>` would work only if a refspec is already configured for tags, which is not the case "out-of-the-box"...)

Anyway, thanks for the pointer.

**Comment by webstech** [comment]
> @phil-blain If you want to give it a try, you will probably want to touch [these lines](https://github.com/gitgitgadget/gitgitgadget/blob/e8ec25f755cc0b770ba956e2932f45b4d0fe0e03/lib/ci-helper.ts#L605-L607).

The simplest way to get the information to 'those lines' may be to return an interface that contains the tag.  The tag is known a couple of layers deeper in `PatchSeries::generateAndSend()`.

Is there another way to do this?  GitHub puts the PR contents at a (somewhat) well known location.  The fetch command could be:
```
git fetch https://github.com/gitgitgadget/git refs/pull/<num>/head:pr<num>
```
to fetch the changes to a local branch (`pr<num>` in this example).

The repo owner may be gitgitgadget or git depending upon where the PR was opened.  The submitted comment could add this info with no change to underlying code.  Is there a risk of missing anything because this does not include `refs/pull/<num>/merge`?

**Comment by phil-blain** [comment]
Hi @webstech!  My idea was more in the context of multiple iterations of a series, where for example I use interactive rebasing to tweak my branch after each iteration and force push to the PR branch, but I'd like to have an easy way to quickly fetch an earlier version of my series, for example to take a look at the `git range-diff` output. 

So in this context `git fetch https://github.com/gitgitgadget/git refs/pull/<num>/head:pr<num>` is not helpful if I already forced-push, because then this ref points to the same commit as my local branch, whereas tags are created by GGG for each sent iteration of a series so they are a fixed reference point on which to compare my work. Does that make sense ?

**Comment by webstech** [comment]
>> Does that make sense ?

Yes.  I had not understood the use case.  Is the `Fetch-It-Via` footer format okay or would you prefer something different?  I have a branch ready to go.

```
Submitted as [pull.59.git.1596151495333.user@example.com](https://lore.kernel.org/git/pull.59.git.1596151495333.user@example.com)

Fetch-It-Via: git fetch https://github.com/gitgitgadget/git pr-59/somebody/master-v1
```

**Comment by phil-blain** [comment]
As I said above, this fetches the tag into FETCH_HEAD but does not keep a local reference, so that's why I suggested this:

~~~
To fetch this version into FETCH_HEAD:

```
git fetch https://github.com/gitgitgadget/git  <ggg-tag-name>
```
To fetch this version to a local tag:

```
git fetch https://github.com/gitgitgadget/git tag <ggg-tag-name>
```
~~~

**Comment by webstech** [comment]
Sorry, missed seeing that as a format suggestion.  If a user does not know what to do with `FETCH_HEAD` the basic fetch instruction is not very helpful.  Assuming most (git-for-windows) users are still learning, would it be better to offer these alternatives and ignore the first?
```
Create a branch using the remote tag name:
git fetch https://github.com/gitgitgadget/git tag <ggg-tag-name>
Create a branch name of your preference:
git fetch https://github.com/gitgitgadget/git <ggg-tag-name>:<local-branch-name>
```

**Comment by phil-blain** [comment]
> Sorry, missed seeing that as a format suggestion. If a user does not know what to do with `FETCH_HEAD` the basic fetch instruction is not very helpful.

That's true, but then they can learn about it :) 
> Assuming most (github git) users are still learning, would it be better to offer these alternatives and ignore the first?
> 
> ```
> Create a local branch using the remote tag name:
> git fetch https://github.com/gitgitgadget/git tag <ggg-tag-name>

That does not create a local *branch*, it create a local *tag*. 

Unfortunately, it also fetches more than just `<ggg-tag-name>`, i.e. if you have other old local or remote branches corresponding to GGG submissions, GGG tags for these will aslo be fetched; `--no-tags` must be added to prevent that.

> Create a local branch name of your preference:
> ```
> git fetch https://github.com/gitgitgadget/git <ggg-tag-name>:<local-branch-name>
> ```

This does not work as it tries to write a tag object into `refs/heads/<local-branch-name>`, which is forbidden:
```
$ git fetch https://github.com/gitgitgadget/git pr-691/phil-blain/list-guides-in-git.1-v1:test-tag --no-tags
error: cannot update ref 'refs/heads/test-tag': trying to write non-commit object e48535af51c3c44ae1f5c15ff63f0abc7f28f0d6 to branch 'refs/heads/test-tag'
From https://github.com/gitgitgadget/git
 ! [new tag]               pr-691/phil-blain/list-guides-in-git.1-v1 -> test-tag  (unable to update local ref)
```
To get a local branch, you need to directly fetch the hash of the commit pointed to by the tag  (see below).

--- 

So a more complete list of options would be:
-```
To fetch this version into `FETCH_HEAD`:
```
git fetch https://github.com/gitgitgadget/git <ggg-tag-name>
```

To fetch this version to local tag `<ggg-tag-name>`:
```
git fetch --no-tags https://github.com/gitgitgadget/git tag <ggg-tag-name> 
```

To fetch this version to local tag `<local-tag>`:
```
git fetch --no-tags https://github.com/gitgitgadget/git <ggg-tag-name>:refs/tags/<local-tag>
```

To fetch this version to local branch `<local-branch>`:
```
git fetch --no-tags https://github.com/gitgitgadget/git <tip-commit-hash>:<local-branch> 
```

-```

but then this might be is a little overkill :P @dscho, @webstech  what do you think ?

**Comment by webstech** [comment]
@phil-blain Sorry for not double checking what you can do with a tag (ie not get it directly to a branch).  I will change to your first two suggestions and open the pull request.

**Comment by phil-blain** [comment]
This was implemented in #295.
